streams/shared/src/main/scala/zio/stream/ZStreamChunk.scala

```diff
@@ -64,6 +64,16 @@ trait ZStreamChunk[-R, +E, @specialized +A] { self =>
         }
     })
 
+  final def collect[B](p: PartialFunction[A, B]): ZStreamChunk[R, E, B] =
+    ZStreamChunk(new ZStream[R, E, Chunk[B]] {
+      override def fold[R1 <: R, E1 >: E, B1 >: Chunk[B], S]: Fold[R1, E1, B1, S] =
+        ZManaged.succeedLazy { (s, cont, f) =>
+          self.fold[R1, E1, A, S].flatMap { fold =>
```

Comment:
This would unfortunately destroy the chunk structure. Could you try rewriting it with `self.chunks.fold` and running `collect(p)` on each chunk?

Reply:
`ZStreamChunk(self.chunks.map(e => e.collect(p))`, @iravid any thoughts about this one?

Reply:
Yes, that'd be nice and short!